### PR TITLE
refactor: [M3-5457] - Make HelpIcon more generic

### PR DESCRIPTION
**Pull request by carrillo-erik**
## Description 📝

Rename the `<HelpIcon />` component to `<TooltipIcon />`.

Replace the former `isError` prop which was a boolean value, with the new `status` prop which allows a constant value to be passed in and render the appropriate SVG icon. The values accepted are `SUCCESS`, `ERROR`, `WARNING`, `INFO`, and `HELP`.

Also, replace CSS classes with the introduction of the `sxTooltipIcon` prop which takes advantage of the new MUI v5 `sx` prop to define specific styling.

## Preview 📷

https://user-images.githubusercontent.com/119514965/228618002-3827c525-9e14-40a3-9fee-23c9986f1cb9.mov

## How to test 🧪
On your local instance of Cloud Manager, visit the the billing landing page and hit the 'Add Payment Method' button.
Next, open the Browser DevTools and go to the 'Components' tab. Select either the GoogleChip or PayPal chip component and toggle it's `state` from `false` to `true` and inspect that an red error icon is rendered.

Additionally, inspect that all former help icons are rendered as expected.

### CODE REVIEW

File: packages/manager/src/components/TooltipIcon/TooltipIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+export type TooltipIconStatus = ObjectValues<typeof TOOLTIP_ICON_STATUS>;
+
+interface Props
+  extends Omit<TooltipProps, 'leaveDelay' | 'title' | 'children'> {
+  sx?: SxProps;
+  sxTooltipIcon?: SxProps;
+  text: string | JSX.Element;
+  className?: string;
+  interactive?: boolean;
+  status: TooltipIconStatus;
```
Comment (bnussman-akamai) [inline packages/manager/src/components/TooltipIcon/TooltipIcon.tsx:31]: What benefit does the `TOOLTIP_ICON_STATUS` object give us compared to just using a string literal for the `status`?

```suggestion
  status: 'success' | 'error' | 'warning' | 'info' | 'help'
```

or 

```typescript
type TooltipStatus = 'success' | 'error' | 'warning' | 'info' | 'help';

interface Props {
  ...
  status: TooltipStatus;
}
```

Comment (carrillo-erik) [reply packages/manager/src/components/TooltipIcon/TooltipIcon.tsx:31]: For starters, it provides auto-completion. Also, it prevents someone from passing a random or misspelled `status`.

**Review by bnussman-akamai — COMMENTED** [review]
I'm seeing some color issues in dark mode 🎨

| `develop` | This PR |
|--------|-------|
| Screenshot 2023-03-30 at 6 38 31 PM      | Screenshot 2023-03-30 at 6 38 47 PM     |

**Comment by carrillo-erik** [comment]
@bnussman-akamai I've addressed the issue for dark mode styles, thank you for verifying this.

**Comment by jaalah-akamai** [comment]
Let's wait until https://github.com/linode/manager/pull/8933 is merged before merging this

File: packages/manager/src/features/Databases/DatabaseDetail/DatabaseSummary/DatabaseSummaryConnectionDetails.tsx [hunk 2]
```diff
@@ -215,9 +211,13 @@ export const DatabaseSummaryConnectionDetails = (props: Props) => {
         Download CA Certificate
       </Button>
       {disableDownloadCACertificateBtn ? (
-        <HelpIcon
-          className={classes.helpIcon}
+        <TooltipIcon
+          sxTooltipIcon={{
+            padding: 0,
+            marginLeft: '4px',
```
Comment (jaalah-akamai) [inline packages/manager/src/features/Databases/DatabaseDetail/DatabaseSummary/DatabaseSummaryConnectionDetails.tsx:217]: This style can be re-used for all instances in this file and also you can remove commented code

**Review by jaalah-akamai — APPROVED** [review]
Solid changes! Approved pending my one comment

File: packages/manager/src/components/TooltipIcon/TooltipIcon.tsx [hunk 3]
```diff
@@ -56,28 +45,55 @@ const HelpIcon: React.FC<CombinedProps> = (props) => {
...
-        {isError ? <ErrorOutline /> : <HelpOutline />}
+      <IconButton data-qa-help-button size="large" sx={sxTooltipIcon}>
+        {renderIcon}
       </IconButton>
     </Tooltip>
   );
 };
 
-export default HelpIcon;
+export default TooltipIcon;
```
Comment (bnussman-akamai) [inline packages/manager/src/components/TooltipIcon/TooltipIcon.tsx:99]: Did we decide we want to not default export anymore? @jaalah-akamai @carrillo-erik

**Review by bnussman-akamai — APPROVED** [review]
Nice job!

**Pull request merged**